### ISSUE TITLE
rounding error in dogecoin value?

### ISSUE THREAD

**Issue by dooglus** [dogecoin/dogecoin#134]
I just tried creating a raw transaction that sends a single large input to a new address with 0 fee.  It was rejected with:

```
error: {"code":-22,"message":"TX rejected"}
```

but the transaction was small enough, and with big enough value to qualify as free.

Further investingation showed me that this line was failing:

```
    if (nFees < tx.GetMinFee(1000, true, GMF_RELAY))
        return error("CTxMemPool::accept() : not enough fees");
```

because nFees was -1 and GetMinFee() was returning 0.

Why did it think the included fee was negative?  It turns out the raw transaction wasn't specifying the output value I asked for.  Here's a minimal example:

```
$ dogecoind decderawtransaction $(dogecoind createrawtransaction '[]' '{"DBwb7y87zGnvmZgnxXdxoVwjbevzNrpVtr":50000000.00000001}') | grep value
        "value" : 50000000.00000002,
```

I ask for a single output with value 50000000.00000001, but get a single output one satoshi bigger than that.  And so the fee is -1 satoshi.

I thought dogecoind used integers for coin values, so am a little confused how this can be happening.  Trying to send 1 satoshi more than 10, 20, 30, or 40 million doges works fine.  It's only 50 million where the problem starts happening.

**Comment by rnicoll** [comment]
Sorry, meant to come back to this - going to add int-based versions of the various JSON RPC calls, rather than either using fixed-precision values in strings or risking weird incompatibility bugs.

**Comment by gigatux** [comment]
You'll definitely make my day if this gets implemented soon! Accounting is quite difficult with these little errors.

**Comment by rnicoll** [comment]
Oh I never checked this was solved - let me know if not, please? Think it was fixed in 1.10 and therefore also 1.14 though
